### ISSUE TITLE
Rewind examples don't compile with Typescript

### ISSUE THREAD

**Issue by jscheid** [ably/ably-js#694]
In your docs you have examples such as [this one](https://github.com/ably/docs/blob/baa0a4666079abba3a3e19e82eb99ca8b8a735d0/content/realtime/channels.textile#setting-channel-options):

```js
const realtime = new Ably.Realtime('{{API_KEY}}');
const channelOptions = {
  params: {
    delta: 'vcdiff',
    rewind: '1'
  }
};
const channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}', channelOptions);
```

But your Typescript types [require all of `params`, `cipher`, and `mode` to be set](https://github.com/ably/ably-js/blob/1aa8bfcb8f04b579c74303ebd6abbe8347cb35dd/ably.d.ts#L255-L259) and so you end up with an error like

```
Argument of type '{ params: { rewind: string; }; }' is not assignable to parameter of type 'ChannelOptions'.
  Type '{ params: { rewind: string; }; }' is missing the following properties from type 'ChannelOptions': cipher, modes [2345]
```

**Comment by SimonWoolf** [comment]
Thanks for the bug report. Fixed by https://github.com/ably/ably-js/pull/695

**Comment by owenpearson** [comment]
@jscheid We've released the fixed type definitions as [1.2.4](https://github.com/ably/ably-js/releases/tag/1.2.4)
